vehicle/mkvi/software/dashboard/dashboard.py

```diff
@@ -0,0 +1,220 @@
+import threading
+import time
+import can
+import cantools
+import RPi.GPIO as GPIO
+
+# bustype = "slcan"          idk lol
+# channel = "/dev/ttyACM1"   which port on the raspi
+# bitrate = 500000           number of bits per second
+
+# Loading in everyone's compiled config files
+dbc_file = "mkv.dbc"
+db = cantools.database.load_file(dbc_file)
+
+# Creating the STATES_DICTIONARY
+STATES_DICTIONARY = {}
+
+
+def init_can(channel, bustype, bitrate, callback, dbc):
+    """Initialize CAN hardware and create CAN database from DBC"""
+    can_bus = can.interface.Bus(
+        channel=channel,
+        bustype=bustype,
+        bitrate=bitrate,
+    )
+
+    kill_flag = threading.Event()
+    listener = threading.Thread(
+        target=dashboard_listener,
+        name="listener",
+        kwargs={"can_bus": can_bus, "callback": callback, "kill_flag": kill_flag},
+    )
+
+    listener.start()
+    return can_bus, kill_flag
+
+
+def dashboard_listener(can_bus, callback, kill_flag):
+    """Thread that runs all the time to listen to CAN messages
+    References:
+      - https://python-can.readthedocs.io/en/master/interfaces/socketcan.html
+      - https://python-can.readthedocs.io/en/master/
+    """
+    while not kill_flag.is_set():
+        msg = can_bus.recv(1)  # 1 second receive timeout
+        if msg:
+            callback(msg, db)
+
+    can_bus.shutdown()
+    print("Exited gracefully")
+
+
+def dashboard_callback(msg, db):
+    """
+    Callback when a CAN message is received, updates appropriate vehicle dictionaries
+    Args:
+        msg (can.Message): CAN message that was received
+        db (cantools.database): Database generated from our DBC
+    """
+
+    try:
+        message = db.decode_message(msg.arbitration_id, msg.data)
+
+    except Exception as _:
```

Comment:
This was one of the functions that I copied pretty much verbatim from your rx_callback() function -- should I keep the try and except in there so that if something accidentally goes wrong it doesn't stop the bus? I'm also not sure what exceptions there could be/how we would figure that out.